src/Microsoft.OpenApi.OData.Reader/Operation/EdmOperationOperationHandler.cs

```diff
@@ -172,6 +172,28 @@ protected override void SetResponses(OpenApiOperation operation)
             }
             else
             {
+                OpenApiSchema schema;
+                if (EdmOperation.ReturnType.TypeKind() == EdmTypeKind.Collection)
+                {
+                    // Get the entity type of the previous segment
+                    IEdmEntityType entityType = Path.Segments.Reverse().Skip(1).Take(1).FirstOrDefault().EntityType;
+                    schema = new OpenApiSchema
+                    {
+                        Title = $"Collection of {entityType.Name}",
+                        Type = "object",
+                        Properties = new Dictionary<string, OpenApiSchema>
+                        {
+                            {
+                                "value", Context.CreateEdmTypeSchema(EdmOperation.ReturnType)
+                            }
+                        }
+                    };
+                }
```

Comment:
It looks you can move these codes into "CreateEdmTypeSchema(operation.ReturnType)".

I am ok now to move it on now.

Reply:
I wasn't sure whether the same applies to collection-valued properties. See: https://docs.oasis-open.org/odata/odata-openapi/v1.0/cn01/odata-openapi-v1.0-cn01.html#sec_CollectionValuedProperties